src/NumberGuessingGame.js

```diff
@@ -1,77 +1,117 @@
-import React, { Component } from "react";
+import React, { Component, useState } from "react";
 import GuessControl from "./GuessControl";
 import GuessMessage from "./GuessMessage";
 import GameOver from "./GameOver";
 
-/**
- *
- * Returns a random integer number from 1-100 inclusive
- */
-function getRandomNumber() {
-  return Math.floor(Math.random() * 100) + 1;
-}
 
-const MAX_ATTEMPTS = 5;
 
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
+const NumberGuessingGame = () => {
+  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber());
+  const [latestGuess, setLatestGuess] = useState(null);
+  const [numberOfGuesses, setNumberOfGuesses] = useState(0);
 
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    };
-
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
+  function getRandomNumber() {
+    return Math.floor(Math.random() * 100) + 1;
   }
 
-  handleGuess(guess) {
-    this.setState({
-      latestGuess: guess,
-      numberOfGuesses: this.state.numberOfGuesses + 1,
-    });
-  }
 
-  handleReset() {
-    this.setState({
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    });
+  const MAX_ATTEMPTS = 5;
+
+  const isCorrectGuess = latestGuess === numberToGuess;
+  const isGameOver = isCorrectGuess || numberOfGuesses === MAX_ATTEMPTS;
+
+  function handleGuess(guess) {
+
+    (Number(guess));
```

Comment:
Great job! Here we need to `setLatestGuess(Number(guess))` otherwise we don't get the hints!